zerver/views/webhooks/hellosign.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from six import text_type
+from six.moves import range
+from typing import Dict, Any, Optional, Union
+
+def format_body(signatories, model_payload):
+    # type: (List[Dict[str, Any]], Dict[str, Any]) -> Any
+    RESULT_TEMPLATE = "The {contract_title}"
+    for i, signatory in enumerate(signatories):
+        identity = str(i) # type: str
+        if signatory['status_code'] == 'awaiting_signature':
+            RESULT_TEMPLATE += ''.join([" is awaiting the signature of {name_",
```

Comment:
You are still using `+` on strings. What I meant was that first create the result in an array and then join it. Example:
```Python
result = []
for i, number in enumerate(['one, 'two', 'three']):
    result.append("{index}: {number}".format(index=i, number=number))

print(''.join(result))
```